sdk/tests/core/test_az_http.c

```diff
@@ -560,22 +560,87 @@ static void test_http_response(void** state)
     }
   }
 
-  az_span response_span = AZ_SPAN_FROM_STR( //
-      "HTTP/1.1 200 Ok\r\n"
-      "Content-Type: text/html; charset=UTF-8\r\n"
-      "\r\n"
-      // body
-      EXAMPLE_BODY);
-
-  az_http_response response = { 0 };
-  az_result const result = az_http_response_init(&response, response_span);
-  assert_true(result == AZ_OK);
-
-  // add parsing here
+  // get body directly
   {
-    az_span body;
-    TEST_EXPECT_SUCCESS(az_http_response_get_body(&response, &body));
-    assert_true(az_span_is_content_equal(body, AZ_SPAN_FROM_STR(EXAMPLE_BODY)));
+    az_span response_span = AZ_SPAN_FROM_STR( //
+        "HTTP/1.1 200 Ok\r\n"
+        "Content-Type: text/html; charset=UTF-8\r\n"
+        "\r\n"
+        // body
+        EXAMPLE_BODY);
+
+    az_http_response response = { 0 };
+    az_result const result = az_http_response_init(&response, response_span);
+    assert_true(result == AZ_OK);
+
+    // get body directly
+    {
+      az_span body;
+      TEST_EXPECT_SUCCESS(az_http_response_get_body(&response, &body));
+      assert_true(az_span_is_content_equal(body, AZ_SPAN_FROM_STR(EXAMPLE_BODY)));
+    }
+  }
+
+  // Bad response. Will get first part right and will block reading beyond the digits
+  {
+    az_span response_span = AZ_SPAN_FROM_STR( //
+        "HTTP/");
+
+    az_http_response response = { 0 };
+    az_result const result = az_http_response_init(&response, response_span);
+    assert_true(result == AZ_OK);
+
+    // read a status line
+    {
+      az_http_response_status_line status_line = { 0 };
+      az_result get_result = az_http_response_get_status_line(&response, &status_line);
+      assert_true(get_result == AZ_ERROR_HTTP_CORRUPT_RESPONSE_HEADER);
+    }
+  }
+
+  // Bad response. handle unexpected end when getting headers
+  {
+    az_span response_span = AZ_SPAN_FROM_STR( //
+        "HTTP/2.0 205 \r\n");
+
+    az_http_response response = { 0 };
+    az_result const result = az_http_response_init(&response, response_span);
+    assert_true(result == AZ_OK);
+
+    // read a status line
+    {
+      az_http_response_status_line status_line = { 0 };
+      az_result get_result = az_http_response_get_status_line(&response, &status_line);
+      assert_true(get_result == AZ_OK);
+      az_span header_name = { 0 };
+      az_span header_value = { 0 };
+      get_result = az_http_response_get_next_header(&response, &header_name, &header_value);
+      assert_true(get_result == AZ_ERROR_HTTP_CORRUPT_RESPONSE_HEADER);
```

Comment:
Same nit here/elsewhere:
```suggestion
      assert_int_equal(get_result, AZ_ERROR_HTTP_CORRUPT_RESPONSE_HEADER);
```